core/src/ast/data.rs

```diff
@@ -150,17 +152,48 @@ impl<V: UsesLifetimes, F: UsesLifetimes> UsesLifetimes for Data<V, F> {
 }
 
 /// Equivalent to `syn::Fields`, but replaces the AST element with a generic.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub struct Fields<T> {
     pub style: Style,
     pub fields: Vec<T>,
+    // This field is intentionally private, to make it easier to add new fields to this struct
+    // in the future. Having a private field prevents the construction of fields by hand like this
+    //
+    // Fields {
+    //     style: Style::Struct,
+    //     fields: vec![],
+    // }
+    //
+    // and you have to use a function instead.
+    // This field can be made public, once darling reaches 1.0.
+    //
+    // This span should span around all fields including the braces.
+    span: Option<Span>,
 }
 
 impl<T> Fields<T> {
+    /// Creates a new [`Fields`] struct.
+    pub fn new(style: Style, fields: Vec<T>) -> Self {
+        Self {
+            style,
+            fields,
+            span: None,
+        }
+    }
+
+    /// Adds a [`Span`] to [`Fields`].
```

Comment:
How are these linking to something?

Reply:
There was an RFC, that added this feature
https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#additions-to-the-documentation-syntax

Reply:
That RFC hasn't been implemented yet: https://github.com/rust-lang/rust/issues/43466